katacomb/katacomb/util/image_util.py

```diff
@@ -88,7 +86,15 @@ def write_image(input_file, output_file, width=1024, height=768, dpi=DEFAULT_DPI
         data = hdus[0].data[ax_select]
         vmin, vmax = zscale.zscale(zscale.sample_image(data))
         image_height, image_width = data.shape
-        fig, ax = _prepare_axes(wcs.WCS(hdus[0]), width, height, image_width, image_height, dpi, slices, trim)
+        # Work out bounding box surrounding finite data
+        # Plot the lot if any axis is completely blanked
+        finite_data = np.where(np.isfinite(data))
+        bbox = (0, image_width, 0, image_height)
+        if data[finite_data].size > 0:
```

Comment:
That's quite an expensive way to check if there are any finite pixels - it's going to create a new array with all the finite pixel values. I'd suggest
```suggestion
        if finite_data[0].size > 0:
```